homeassistant/components/rainforest_eagle/config_flow.py

```diff
@@ -7,20 +7,28 @@
 import voluptuous as vol
 
 from homeassistant import config_entries
-from homeassistant.const import CONF_TYPE
+from homeassistant.const import CONF_HOST, CONF_TYPE
 from homeassistant.data_entry_flow import FlowResult
 
 from . import data
 from .const import CONF_CLOUD_ID, CONF_HARDWARE_ADDRESS, CONF_INSTALL_CODE, DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_CLOUD_ID): str,
-        vol.Required(CONF_INSTALL_CODE): str,
-    }
-)
+
+def create_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Create user schema with passed in defaults if available."""
+    if user_input is None:
+        user_input = {}
+    return vol.Schema(
+        {
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST)): str,
```

Comment:
Does a default `None` work with the string validation?

Reply:
It's required so the default is never entered. The default is used here only to provide a value in the frontend.